src/fitparser/index.ts

```diff
@@ -307,6 +312,88 @@ export class FitParser {
         logger.info("FitParser.parse", logHelper.user(user), logHelper.fitFileActivity(fitFileActivity), `Data: ${logFields.join(", ")}`)
     }
 
+    private intensity_prefixes = {
+        "warmup": "🟢 Warmup: ",
+        "cooldown": "🟡 Cooldown: ",
+        "interval": "🚀 Interval: ",
+        "rest": "🌿 Rest: ",
+    }
+
+    private formatWorkoutSteps(workout_steps: any): string {
+        let formatted_step_lines = []
+
+        for (let step of workout_steps) {
+            let line = this.intensity_prefixes[step.intensity] || ""
+
+            switch (step.duration_type) {
+                case "time":
+                    line += this.formatMsTime(step.duration_value)
+                    break
+                case "repeat_until_steps_cmplt":
+                    let rep_start_index = step.duration_value
+                    let rep_count = step.target_value
+
+                    let rep_line = `🔄 Repeat ${rep_count}x:`
+
+                    for (var i = rep_start_index; i < formatted_step_lines.length; i++) {
+                        formatted_step_lines[i] = "   - " + formatted_step_lines[i]
+                    }
+
+                    formatted_step_lines.splice(rep_start_index, 0, rep_line)
+                    continue
+            }
+            
+            switch (step.target_type) {
+                case "speed":
+                    if (step.target_value) {
+                        line += ` @ ${this.formatSpeedAsPace(step.target_value)}`
```

Comment:
this might be too specific to running, we might want to format to a speed that makes sense for cycling workouts as well and probably add support for power and heart rate target types as well